src/source/mod.rs

```diff
@@ -637,7 +637,7 @@ fn decode_with_charset(
   let text = match text_encoding::convert_to_utf8(bytes.as_ref(), charset)? {
     Cow::Borrowed(text) => {
       if text.starts_with(text_encoding::BOM_CHAR) {
-        text[..text_encoding::BOM_CHAR.len_utf8()].to_string()
+        text[text_encoding::BOM_CHAR.len_utf8()..].to_string()
```

Comment:
I also found this critical bug fix in my working tree that I forgot to commit, so good this happened